core/src/main/java/org/infinispan/topology/ClusterTopologyManager.java

```diff
@@ -37,23 +39,17 @@ boolean isCoordinator() {
     * If the node is the first to join the cache, the returned topology does include the joiner,
     * and it is never {@code null}.
     */
-   CacheStatusResponse handleJoin(String cacheName, Address joiner, CacheJoinInfo joinInfo, int viewId) throws Exception;
+   CompletionStage<CacheStatusResponse> handleJoin(String cacheName, Address joiner, CacheJoinInfo joinInfo, int viewId) throws Exception;
 
    /**
     * Signals that a member is leaving the cache.
     */
-   void handleLeave(String cacheName, Address leaver, int viewId) throws Exception;
+   CompletionStage<Void> handleLeave(String cacheName, Address leaver, int viewId) throws Exception;
 
    /**
     * Marks the rebalance as complete on the sender.
     */
-   void handleRebalancePhaseConfirm(String cacheName, Address node, int topologyId, Throwable throwable, int viewId) throws Exception;
-
-   void broadcastRebalanceStart(String cacheName, CacheTopology cacheTopology, boolean totalOrder, boolean distributed);
```

Comment:
Because `ClusterCacheStatus` is not a separate component, so it doesn't have to go through the interface. I just noticed that `broadcastShutdownCache()` was in the same boat, so I removed it as well.